src/components/Toolbar.tsx

```diff
@@ -70,6 +73,32 @@ async function showGitOperationDialog(
   }
 }
 
+async function showTagListDialog(
+  model: IGitExtension,
+  repo: IToolbarState['repository']
+): Promise<void> {
+  const title = `Tags`;
+  let tagsValue = await showDialog({
+    title: title,
+    body: new GitTagDialog(model, repo),
+    buttons: [Dialog.okButton({ label: 'OK' })]
+  });
+  while (!tagsValue.button.accept) {
+    tagsValue = await showDialog({
+      title: title,
+      body: new GitTagDialog(model, repo),
+      buttons: [Dialog.okButton({ label: 'DISMISS' })]
+    });
+  }
```

Comment:
```suggestion
  });
```

You should restore the cancel button on the GitTagDialog. For example, the user may want to stop checking out a tag because he clicked not intentionally on the tag button.